src/rod/sdf/sdf.py

```diff
@@ -82,13 +79,12 @@ def load(sdf: pathlib.Path | str, is_urdf: bool | None = None) -> Sdf:
             and pathlib.Path(sdf).is_file()
         ):
             sdf_string = pathlib.Path(sdf).read_text(encoding="utf-8")
-            is_urdf = (
-                is_urdf if is_urdf is not None else pathlib.Path(sdf).suffix == ".urdf"
-            )
+            is_urdf = pathlib.Path(sdf).suffix == ".urdf"
 
         # Finally, it must be a SDF/URDF string
         else:
             sdf_string = sdf
+            is_urdf = "sdf" not in sdf_string
```

Comment:
Mmh this seems to me extremely fragile. The sequence of the three characters `s-d-f` could be part of a link name, a joint name, etc of a URDF model. A more robust solution would be looking for the string `<robot>`. This enforces the searched string to be a XML element[^1] that is specific to URDF. There is no `/robot` element supported in the SDF specification.

cc-ing @traversaro here for also having his feedback on the topic.

[^1]: I hope no one is so mentally ill to name e.g. a link as `<robot>`, which is actually a valid link name :smile:

Reply:
Thanks a lot Diego, I agree with you! Addressed in https://github.com/ami-iit/rod/pull/42/commits/4d27396c3151fc7af39b446e3ea43e37115aa5b9